### DISCUSSION TITLE
how to use an external touch library wih tft_espi

### DISCUSSION THREAD

**Discussion by hannescam** [Bodmer/TFT_eSPI#2111] [Q&A - Touch] [answered]
how can is use an external touch library (for XPT2046) with tft_espi
please provide an example code thankyou

**Comment by Bodmer** [comment]
I have already suggested a suitable library for you here:
https://github.com/Bodmer/TFT_eSPI/discussions/2097#discussioncomment-3910728

Have you looked at that? Code is included in the examples that comes with the library!!!

Please only raise "issues" for problems with the TFT_eSPI library, not for gerneral how to questions.

**Comment by Bodmer** [comment, marked as answer]
I had a look at your profile and saw you are 13, so in view of that I will provide more help. The TFT_eTouch library can be found here:
https://github.com/achillhasler/TFT_eTouch

You need to download the zip file and install the library manually in the Arduino IDE. There are online guides that tell you how do this, so Google that.

The touch screeninterface pins need to be connected up to the processor and those pins defined in the example sketches for the library. Library documentation is included in the docs folder. The examples should work fine with little modification. Try them all. I have only briefly used that library when it was first created so am unable to give detailed support in the case of difficulties with it.

**Reply by hannescam** [reply]
it is funny because i saw that library and i thought it was the wrong one but no
and i use platformio with linux and there is it also pretty simple to include a zip library
and thankyou
